README.rst

```diff
@@ -47,34 +63,19 @@ To build a complete set of HTML documentation, you must have Sphinx, which can b
 
 The built html files can be found in doc/_build/html afterward.
 
-
-Twisted
```

Comment:
This section was added in 2014 and the twisted support have been removed since eventlet 0.14.0. We are now close to 2024, I think we can safely remove this section. The message have been passed.